packages/preview-middleware-client/src/utils/version.ts

```diff
@@ -1,15 +1,75 @@
-import type { SingleVersionInfo } from '../../types/global';
 import VersionInfo from 'sap/ui/VersionInfo';
 import Log from 'sap/base/Log';
 
+type SingleVersionInfo =
+    | {
+          name: string;
+          version: string;
+      }
+    | undefined;
+
+export type Ui5VersionInfo = {
+    majorUi5Version: number;
+    minorUi5Version: number;
+};
+
+/**
+ * Default minimal supported UI5 version
+ */
+const minVersionInfo = {
+    majorUi5Version: 1,
+    minorUi5Version: 71
+} as Readonly<Ui5VersionInfo>;
+
 /**
  * Retrieve the UI5 version from sap.ui.core library
+ *
+ * @returns Ui5VersionInfo
  */
 export async function getUi5Version() {
     let version = ((await VersionInfo.load({ library: 'sap.ui.core' })) as SingleVersionInfo)?.version;
     if (!version) {
         Log.error('Could not get UI5 version of application. Using 1.121.0 as fallback.');
         version = '1.121.0';
     }
-    return version;
+    const [major, minor] = version.replace(/snapshot-untested|snapshot-|snapshot/, '').split('.').map(Number);
```

Comment:
Same as the comment above regarding replacement and converting to number.

Reply:
done